velox/functions/prestosql/UpperLower.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/StringWriter.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/lib/StringEncodingUtils.h"
+#include "velox/functions/lib/string/StringImpl.h"
+
+namespace facebook::velox::functions {
+
+/// Function to convert string to upper or lower case. Ascii and unicode
+/// conversion are supported.
+/// @tparam isLower Instantiate for upper or lower.
+/// @tparam forSpark When true, Spark's specific behavior is considered, e.g.
+/// for 'İ' Spark's lower case is 'i̇' and Presto's is 'i'.
+template <bool isLower, bool forSpark>
```

Comment:
Please also add the test when forSpark is false
Do we need to move this file to directory velox/functions/lib

Reply:
The tests of `forSpark` being false are covered by the `StringImplTest`. Do we need to repeat them?

https://github.com/facebookincubator/velox/pull/13158/files#diff-df8512514193690325471bddf44afcd54a1b37493a9ad8b607f8c0d1cd290a95R41-R47

Moved UpperLower.h to the lib, thanks.

Reply:
It does not cover I with dot above case, this is the different code path

Reply:
I got it now.